SDK_ANATOMY.md

```diff
@@ -0,0 +1,330 @@
+
+* `nodes`, which is a list of `Node` objects which gets sorted with the most preferred nodes at the start of the list (the `Node` class has a custom compare method), and this is done to facilitate client-side load balancing of the network.
+
+* `executor`, a reference to the executor which will be used to create channels for the `Node`s (in practice, this is always the `Client`'s executor).
+* `lock`, used to make the `Network` object thread safe.
+
+
+
+`setNetwork()` will update this `Network` to the given list.  It will close a `Node` and remove it from this network if:
+
+* It is not in the given list, or...
+* In the given list the same `AccountId` is mapped to a different `"ipAddress:portNumber"`.
+
+`setNetwork()` will then add nodes from the list.
+
+
+
+`getNodeAccountIdsForExecute()` gets a list of the `AccountId`s for the first (sorted) 1/3rd (rounded up) of healthy nodes in this `Network`.  This is used by  `Query` and `Transaction` to populate their `nodeAccountId`s, lists containing the `AccountId`s of `Node`s that the `Query` or `Transaction` will be attempted with.
+
+
+
+
+
+### `Node`:
+
+This is a connection to one node in the network.  Inherits from `ManagedNode` (which is where much of the meat is).
+
+`Node` has `increaseDelay()` and `decreaseDelay()` methods.  `increaseDelay()` gets called whenever the node fails, and `decreaseDelay()` gets called when it succeeds.  The delay starts at 250 millisecs, and doubles on each `increaseDelay()` up to a limit of 8000.  It halves down to a limit of 250 on each `decreaseDelay()`.  Whenever `increaseDelay()` is called, the node gets marked as unhealthy for the duration of the delay.
+
+`Node` has a custom `compareTo()` method.  If one `Node` is healthy and the other is not, the healthy one is preferred.  Otherwise, the `Node` that's
+been used the least times is preferred.  If they've been used the same number of times, the one that was used longest ago is preferred.
+
+ 
+
+
+
+### `ManagedNode`:
+
+Has an `address`, a `channel`, an `executor` (ultimately from `Client`), `lastUsed` and `useCount`.
+
+`ManagedNode` is inherited by `Node` and `MirrorNode`.
+
+`ManagedNode` keeps track of stats about how this node has been used, and it constructs its channel on demand, which is a `grpc.ManagedChannel`, and which is built as a plaintext channel, using the `executor`, and using the user agent from `getUserAgent()`.
+
+`channel` is a `grpc.ManagedChannel` instead of a normal `grpc.Channel` so that we can customize how it is set up (for example, we can give it the specified executor, and we can shut it down in the desired manner).
+
+`ManagedNode` has the methods `inUse()`, which causes the `ManagedNode` to record that it is being used, `getChannel()`, `close()`, and `getUserAgent()`.
+
+The user agent is a string that is used to identify the client to the server.  In this case, it's `"hedera-sdk-java/v{NUMBER}"`.
+
+
+
+
+
+### `Executable`:
```

Comment:
This is what I was thinking with separating intent from java specific implementation
```markdown
Intent:

Extract common code from all requests that retries the request on different nodes with node deprioritization, exponential back-off, and response code handling.

How we acheive this in the Java SDK:

We create several `abstract` methods which must be overwritten so `Executable` can work correctly.

<list of abstract methods and what they're used for>
```